MarsQACompetitionTaskNUnit/Pages/CertificationPage.cs

```diff
@@ -0,0 +1,218 @@
+
+
+            // Click on save button
+            WaitUtils.WaitToBeVisible(driver, "Xpath", "AddButton", 5);
+            AddButton.Click();
+            Thread.Sleep(1000);
+
+        }
+
+        public bool IsCertificationRecordPresent(string Certificate, string From, string Year, int rowNumber = 0)
+        {
+            bool recordPresent = false;
+            string getCertificate, getFrom, getYear;
+            int certificationRowPresent;
+
+            if (rowNumber == 0)
+            {
+                certificationRowPresent = GetCertificationRow(Certificate, From, Year);
+                if (certificationRowPresent > 0)
+                {
+                    recordPresent = true;
+                }
+            }
+            else
+            {
+                WaitUtils.WaitToBeVisible(driver, "Xpath", "getCertificate", 60);
+                getCertificate = driver.FindElement(By.XPath($"//div[@data-tab='fourth']//table/tbody[{rowNumber}]/tr/td[1]")).Text;
+                WaitUtils.WaitToBeVisible(driver, "Xpath", "getFrom", 60);
+                getFrom = driver.FindElement(By.XPath($"//div[@data-tab='fourth']//table/tbody[{rowNumber}]/tr/td[2]")).Text;
+                getYear = driver.FindElement(By.XPath($"//div[@data-tab='fourth']//table/tbody[{rowNumber}]/tr/td[3]")).Text;
+               
+                ReportLogger.LogInfo($"Retrieved [row {rowNumber}]: Country: {getCertificate}");
+                if (Certificate.Equals(getCertificate) && From.Equals(getFrom) && Year.Equals(getYear)) { recordPresent = true; }
+            }
+
+            return recordPresent;
+        }
+
+        public int RowCount()
+        {
+            return CertificationRows.Count;
+        }
+
+        public int GetCertificationRow(string Certificate, string From, string Year)
+        {
+            string getCertificate, getFrom, getYear;
+
+            for (int i = 1; i <= RowCount(); i++)
+            {
+                try
+                {
+                    getCertificate = driver.FindElement(By.XPath($"//div[@data-tab='fourth']//table/tbody[{i}]/tr/td[1]")).Text;
+                    getFrom = driver.FindElement(By.XPath($"//div[@data-tab='fourth']//table/tbody[{i}]/tr/td[2]")).Text;
+                    getYear = driver.FindElement(By.XPath($"//div[@data-tab='fourth']//table/tbody[{i}]/tr/td[3]")).Text;
+                    if (Certificate.Equals(getCertificate) && From.Equals(getFrom) && Year.Equals(getYear))
+                    {
+                        ReportLogger.LogInfo($"Certification Present at row: {i}");
+                        return i;
+                    }
+                }
+                catch (NoSuchElementException)
+                {
+                    continue;
+                }
+            }
+            return 0;
+        }
+
+
+        public void EditCertificationRecord(string Certificate, string From, string Year)
```

Comment:
also i didnt see what you mentioned in slack can you mention what test this is?

Reply:
TestEditCertificationRecordWithNullData